src/main/java/frc/robot/CommandFactory.java

```diff
@@ -37,6 +40,65 @@ public CommandFactory(
         this.superStructure = superStructure;
     }
 
+    private boolean atTargetPosition(double currentPosition, double targetPosition, double threshold) {
+        return Math.abs(currentPosition - targetPosition) <= threshold;
+    }
+
+    /**
+     * Tells the super Structure where to go based on the Robot Mode
+     *
+     * @param robotMode A supplier for the current RobotMode so it knows where to go
+     * @return The command to follow the current position based on the Robot Mode
+     */
+    public Command goToPosition(Supplier<RobotMode> robotMode) {
```

Comment:
I think this belongs better in the `SuperStructure`, as it is just acting on the elevator and arm.